### ISSUE TITLE
Implement Injazz HUD mockup

### ISSUE THREAD

**Issue by PJB3005** [space-wizards/space-station-14#272]
![](https://cdn.discordapp.com/attachments/346743671787356160/598475332688936970/UI_-_Wide_-_RC3.png)

Bits:

* [X] Top left window buttons
* [ ] Status effects on the right
  * Health is over there but that's about it. Kinda half assed.
* [x] Inventory Windows
  * Still merged with character window
* [x] Hotbar
* [x] Quick-access inventory slots in the bottom right
* [x] Combat mode toggle.

**Comment by N3X15** [comment]
Looks good, you guys might actually prove me wrong.

**Comment by tentekal** [comment]
Can we also add the Filter Chat button beside the Admin tab? I know I hashed out a really crude scheme for it and injazz made an actual mockup but I can't be arsed to go back into the discord history.

**Comment by PJB3005** [comment]
Implemented the inventory stuff, the rest is gonna require more effort so I'm gonna remove self assignment.

**Comment by chairbender** [comment]
going to work on the status effects part of this

**Comment by metalgearsloth** [comment]
Check in discord for latest for anyone who wants to work on it.

**Comment by Injazz** [comment]
https://www.figma.com/file/KE5eKymegsLrsQdjZGbTIs/UI?node-id=0%3A1
here our figma project, it supports Sketch, SVG, PNG and PDF, also crossplat (web-based), so it's kinda comfy tool to use for mockups
ask me if you want to work on our mockup as well

**Comment by chairbender** [comment]
got busy IRL, should have time now and I think I will take on the status effects stuff again

**Comment by Zumorica** [comment]
Sounds good :+1: 
Having layer support for status effects would be cool.

**Comment by chairbender** [comment]
for reference, layer stuff covered in #2116

**Comment by chairbender** [comment]
top menu's done, going to tackle the inventory slots next

**Comment by chairbender** [comment]
After the inventory stuff is taken care of in #2965 , what else is left to do before we can mark this as done? I've been working on this issue since it's marked as medium priority, I'd like to move on to other priority stuff if all the UI changes we consider to be a priority here have been addressed. From what I can tell from the original description, I've pretty much covered everything.

**Comment by metalgearsloth** [comment]
IMO everything that's a priority is done. With that being said the combat doll probably needs changing in future and also the chatbox (though I don't think we've really agreed on the chatbox so best to leave that).

**Comment by SweptWasTaken** [comment]
> 
> 
> After the inventory stuff is taken care of in #2965 , what else is left to do before we can mark this as done? I've been working on this issue since it's marked as medium priority, I'd like to move on to other priority stuff if all the UI changes we consider to be a priority here have been addressed. From what I can tell from the original description, I've pretty much covered everything.

Nailing yourself to a single task like this is a good way to burn out fast. You've already done a lot, why not go work on something fresh for a bit? 

Personally if I wanted to anything it would be an option for a more verbose chat-window like in the Figma UI. It would probably be a pretty complicated task as it would resize the active game window with the popped out chat like in vanilla SS13.

![image](https://user-images.githubusercontent.com/49448379/104115470-a438e280-5307-11eb-9513-314f73f3c741.png)

**Comment by metalgearsloth** [comment]
> Nailing yourself to a single task like this is a good way to burn out fast. You've already done a lot, why not go work on something fresh for a bit?

Swept speaks the true true.

I preferred the big chat but last time it was brought up on discord afaik a few people didn't like it. The main reason I want it is because the PVS range is a square so it will help align it more with the viewport rather than for any sort of nostalgic reason (at the very least our current chat is way too small for a medium-size server).

**Comment by SweptWasTaken** [comment]
> 
> 
> > Nailing yourself to a single task like this is a good way to burn out fast. You've already done a lot, why not go work on something fresh for a bit?
> 
> Swept speaks the true true.
> 
> I preferred the big chat but last time it was brought up on discord afaik a few people didn't like it. The main reason I want it is because the PVS range is a square so it will help align it more with the viewport rather than for any sort of nostalgic reason (at the very least our current chat is way too small for a medium-size server).

Will the chatbox scale with the UI like that? Like if i'm on 4K could I have a chatbox the side of a webpage?

**Comment by AJCM-git** [comment]
Agree with swept that working on a single thing like this for too long can burn you. But if you aren't feeling burned yet you could implement both chats from the figma mockup and add an option below UI scaling to change between them (Chat style: small/classic or something like that)

**Comment by chairbender** [comment]
Burnout's not gonna be a problem for me but thanks for the suggestions 👍 I'll check in with maintainers what they're thinking about the current chat mockups.

**Comment by SweptWasTaken** [comment]
I'd say this is complete. Closing.
